src/core/openthread.cpp

```diff
@@ -1256,9 +1256,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
     ThreadError error;
+    Message *message;
+    Ip6::Header header;
 
     otLogFuncEntry();
 
+    message = static_cast<Message *>(aMessage);
+    assert(message->GetLength() >= sizeof(header));
+
+    message->Read(0, sizeof(header), &header);
+    assert(!header.GetDestination().IsRealmLocalMulticast() || header.GetNextHeader() != Ip6::kProtoHopOpts);
```

Comment:
I updated this PR to strip the MPL Option when:

1. The MPL Option is the only option and the HBH Options header is only 8 bytes long.
2. The MPL Option is the last option and starts on an 8-byte boundary.

All other conditions, the MPL Option is simply replaced with PadN.  Please review.